tests/nested/manual/core20-gadget-cloud-conf/task.yaml

```diff
@@ -0,0 +1,154 @@
+  cp "$TESTSLIB"/assertions/developer1.account-key "$NESTED_FAKESTORE_BLOB_DIR/asserts"
+
+  # modify and repack gadget snap to add a defaults section and use our own
+  # prepare-device hook to use the fakedevicesvc
+  mkdir "$(nested_get_extra_snaps_path)"
+
+  # Get the snakeoil key and cert for signing gadget assets (shim)
+  KEY_NAME=$(nested_get_snakeoil_key)
+  SNAKEOIL_KEY="$PWD/$KEY_NAME.key"
+  SNAKEOIL_CERT="$PWD/$KEY_NAME.pem"
+
+  snap download --basename=pc --channel="20/edge" pc
+  unsquashfs -d pc-gadget pc.snap
+
+  # delay all refreshes for a week from now, as otherwise refreshes for our
+  # snaps (which are asserted by the testrootorg authority-id) may happen, which
+  # will break things because the signing keys won't match, etc. and 
+  # specifically snap-bootstrap in the kernel snap from the store won't trust
+  # the seed keys to unlock the encrypted data partition in the initramfs
+  sed defaults.yaml -e "s/@HOLD-TIME@/$(date --date='next week' +%Y-%m-%dT%H:%M:%S%:z)/" >> \
+    pc-gadget/meta/gadget.yaml
+
+  # install the cloud.conf
+  cp cloud.conf pc-gadget/cloud.conf
+
+  # TODO: enable this bit when things are ready to use a testkeys signed model
+  # assertion
+  # copy the prepare-device hook to use our fakedevicesvc
+  # mkdir -p pc-gadget/meta/hooks/
+  # cp prepare-device pc-gadget/meta/hooks/
+
+  nested_secboot_sign_gadget pc-gadget "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
+  snap pack pc-gadget/ extra-snaps/
+  rm -rf pc-gadget/
+
+  rm -f "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
+
+  # TODO: enable this when ready, currently serial assertions requests don't 
+  # work with the fakedevicesvc, needs a little bit of work somewhere
+  # start fake device svc
+  # #shellcheck disable=SC2148
+  #  systemd-run --unit fakedevicesvc fakedevicesvc localhost:11029
+
+  "$TESTSTOOLS"/nested-state build-image core
+  # first boot will use seed1 to create the normal-user in addition to the 
+  # system-user assertion
+  "$TESTSTOOLS"/nested-state create-vm core
+
+restore: |
+  if [ "$TRUST_TEST_KEYS" = "false" ]; then
+      echo "This test needs test keys to be trusted"
+      exit
+  fi
+
+  # stop fake device svc
+  # systemctl stop fakedevicesvc
+
+  #shellcheck source=tests/lib/store.sh
+  . "$TESTSLIB"/store.sh
+  teardown_fake_store "$NESTED_FAKESTORE_BLOB_DIR"
+
+execute: |
+  if [ "$TRUST_TEST_KEYS" = "false" ]; then
+      echo "This test needs test keys to be trusted"
+      exit
+  fi
+
+  #shellcheck source=tests/lib/nested.sh
+  . "$TESTSLIB/nested.sh"
+
+  echo "The initial cloud-init user was created"
+  nested_exec "cat /var/lib/extrausers/passwd" | MATCH normal-user
+
+  echo "And we can run things as the normal user"
+  nested_exec_as normal-user ubuntu "sudo true"
+
+  # TODO: is there a better thing we can wait for here instead? maybe the log
+  # message from snapd directly via journalctl ?
+  echo "Waiting for snapd to react to cloud-init"
+  sleep 60
```

Comment:
perhaps you can wait until cloud-init is disabled after running, I mean, retry using the following check

Reply:
yes, but we can't call `nested_exec` from `retry` since it's a bash function, it would be nice if nested-state had an exec command

Reply:
You are right, I'll work in a nested_retry command since it is required in many tests already.

Reply:
Would it maybe make sense to have a manual retry via a loop? Something like:
```sh
diff --git a/tests/nested/manual/core20-gadget-cloud-conf/task.yaml b/tests/nested/manual/core20-gadget-cloud-conf/task.yaml
index ccbc09d4be..26ee51e3a6 100644
--- a/tests/nested/manual/core20-gadget-cloud-conf/task.yaml
+++ b/tests/nested/manual/core20-gadget-cloud-conf/task.yaml
@@ -140,7 +140,12 @@ execute: |
   # TODO: is there a better thing we can wait for here instead? maybe the log
   # message from snapd directly via journalctl ?
   echo "Waiting for snapd to react to cloud-init"
-  sleep 60
+  for i in $(seq 60); do
+      if nested_exec "cloud-init status" | grep "status: disabled"; then
+          break
+      fi
+      sleep 1
+  done
 
   echo "Ensuring that cloud-init got disabled after running"
   nested_exec "cloud-init status" | MATCH "status: disabled"
```
Or do we know that it always takes 60s? Or am I missing something else?

Reply:
well the issue is that we aren't waiting for cloud-init, we are waiting for snapd to _react_ to cloud-init, if I went with your proposal to do a loop I would need something like

```bash
  for i in $(seq 60); do
      if nested_exec "journalctl --no-pager -u snapd" | grep "Cloud-init reported as blah blah blah"; then
          break
      fi
      sleep 1
  done
```

also to be clear, this was just cargo culted from the all the other cloud-init nested tests, this is not a new pattern I am introducing here, if folks really feel strongly about not having this sleep then we should get rid of all the other sleeps for cloud-init we have elsewhere, it just takes work and time to get the bits right hence why I haven't done it yet. That and given that nested tests are already excruciatingly slow, another 1 minute doesn't seem to be that noticable IMHO.

Reply:
I pushed a change like I suggested without the sleep, let's see how it does, it passed in one test for me

Reply:
The patch seems to work, so this PR is hopefully unblocked